servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultSingleAddressHttpClientBuilder.java

```diff
@@ -219,6 +234,32 @@ private DefaultSingleAddressHttpClientBuilder(@Nullable final U address,
                     proxyAddress != null ? proxyAddress : builder.address);
         }
 
+        Publisher<? extends ServiceDiscovererEvent<R>> discover(final SdStatusCompletable sdStatus) {
+            assert builder.address != null : "Attempted to buildStreaming with an unknown address";
+            final BiIntFunction<Throwable, ? extends Completable> retryWhen = builder.serviceDiscovererRetryStrategy;
+            return builder.serviceDiscoverer.discover(proxyAddress != null ? proxyAddress : builder.address)
+                    .retryWhen((i, t) -> {
+                        sdStatus.nextError(t);
+                        return retryWhen.apply(i, t);
+                    })
+                    .whenFinally(new TerminalSignalConsumer() {
+                        @Override
+                        public void onComplete() {
+                            sdStatus.onComplete();
```

Comment:
As mentioned in the `AutoRetryStrategy` comment, we should not send an `onComplete()` here as it may prematurely retry a request. Instead let the LB decide when to retry if SD completes.